src/neurostatslib/noise_model.py

```diff
@@ -0,0 +1,320 @@
+"""Noise model classes for GLMs."""
+
+import abc
+from typing import Callable, Union
+
+import jax
+import jax.numpy as jnp
+
+from .base_class import _Base
+
+KeyArray = Union[jnp.ndarray, jax.random.PRNGKeyArray]
+
+__all__ = ["PoissonNoiseModel"]
+
+
+def __dir__():
+    return __all__
+
+
+class NoiseModel(_Base, abc.ABC):
+    """
+    Abstract noise model class for neural data processing.
+
+    This is an abstract base class used to implement noise models for neural data.
+    Specific noise models that inherit from this class should define their versions
+    of the abstract methods: negative_log_likelihood, emission_probability, and
+    residual_deviance.
+
+    Attributes
+    ----------
+    FLOAT_EPS :
+        A small value used to ensure numerical stability. Set to the machine epsilon for float32.
+    inverse_link_function :
+        A function that transforms a set of predictors to the domain of the model parameter.
+
+    See Also
+    --------
+    [PoissonNoiseModel](./#neurostatslib.noise_model.PoissonNoiseModel) : A specific implementation of a
+    noise model using the Poisson distribution.
+    """
+
+    FLOAT_EPS = jnp.finfo(jnp.float32).eps
+
+    def __init__(self, inverse_link_function: Callable, **kwargs):
+        super().__init__(**kwargs)
+        self._check_inverse_link_function(inverse_link_function)
```

Comment:
the setter is not initialized yet, once __init__ returns it would be called, but within the __init__ call the check won't be perfomed. if you comment out the line, it fails

(TestPoissonNoiseModel.test_initialization_link_is_callable[1])